src/main/java/frc/robot/subsystems/ArmSubsystem.java

```diff
@@ -48,17 +51,58 @@ public ArmSubsystem(MoInput input) {
         armChooser.addOption("Direct (PID)", pidArmControlCommand);
 
         MoShuffleboard.getInstance().matchTab.add("Arm Control Mode", armChooser).withWidget(BuiltInWidgets.kComboBoxChooser);
+ 
+        this.zero();
+        MoShuffleboard.getInstance().settingsTab
+            .add("Recalculate Arm Position", new InstantCommand(this::zero));
     }
 
     @Override
     public void periodic() {
         this.updateDefaultCommand();
     }
 
+    public void zero() {
+        this.leftShoulder.getEncoder().setPosition(
+            ((this.leftShoulder.getAbsoluteEncoder(SparkMaxAbsoluteEncoder.Type.kDutyCycle)
+                .getPosition() - MoPrefs.absShoulderZero.get() + 1) % 1) * MoPrefs.shoulderAbsToRel.get()
+        );
+        this.wrist.getEncoder().setPosition(
+            ((this.wrist.getAbsoluteEncoder(SparkMaxAbsoluteEncoder.Type.kDutyCycle)
+                .getPosition() - MoPrefs.absWristZero.get() + 1) % 1) * MoPrefs.wristAbsToRel.get()
+        );
```

Comment:
Again, two comments here:
1. This is doing too much in too few lines, making it tough to read. Split the calculations out onto multiple lines.
2. You should just save a reference to the encoders when the class is initialized. Instead of using `this.shoulder.getEncoder()` or `this.shoulder.getAbsoluteEncoder(SparkMaxAbsoluteEncoder.Type.kDutyCycle`, we could just use `this.shoulderEncoder` or `this.shoulderAbsEncoder`

Something like
```java
double shoulder = this.shoulderAbsEncoder.getPosition();

shoulder = (shoulder + 1 - MoPrefs.absShoulderZero.get()) % 1;

shoulder *= moPrefs.shoulderAbsToRel.get();

this.shoulderEncoder.setPosition(shoulder);
```